### PR TITLE
009 role ordering

### PR DESCRIPTION
**Pull request by ldevlin559**

### CODE REVIEW

**Review by TaylorB-softdevkainos — APPROVED** [review]

File: src/main/java/org/example/services/JobRoleService.java [hunk 1]
```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
...
-        }
+            final int page, final int pageSize, final String fieldName,
+            final String orderBy) throws SQLException {
+        final int offset = (page - 1) * pageSize;
+        return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
+                jobRoleDao.getOpenJobRoles(offset, pageSize,
+                        fieldName, orderBy));
+}
+
+    public String getCurrentFieldFilter(final String fieldName)
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: I dont see the point in this Luke - you pass in the fieldName just to pass it back out ?

File: src/main/java/org/example/services/JobRoleService.java [hunk 2]
```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
...
+            throws SQLException {
+        return fieldName;
     }
+
+    public String getCurrentOrderByFilter(final String orderBy)
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: I dont see the point in this Luke - you pass in the orderBy just to pass it back out ?

File: src/main/java/org/example/services/JobRoleService.java [hunk 3]
```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
...
+    }
+
+
     public int getTotalpages(final int pageSize, final int page)
             throws SQLException {
         int totalRecords = jobRoleDao.getTotalOpenJobs();
         return (int) Math.ceil((double) totalRecords / pageSize);
     }
+
+    public int getCurrentPage(final int currentPage)
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: You can delete this method Luke, its not being used anywhere

File: src/main/java/org/example/services/JobRoleService.java [hunk 4]
```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
...
+            throws SQLException {
+        return currentPage;
+    }
+
+    public int getNextPage(final int currentPage)
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: You can delete this method Luke, its not being used anywhere

File: src/main/java/org/example/services/JobRoleService.java [hunk 5]
```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
...
+            throws SQLException {
+        final int pageSize = 10;
+        if (currentPage >= getTotalpages(pageSize, currentPage)) {
+            return currentPage;
+        } else {
+            return currentPage + 1;
+        }
+    }
+
+    public int getPreviousPage(final int currentPage)
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: You can delete this method Luke, its not being used anywhere

File: src/main/java/org/example/controllers/JobRoleController.java [hunk 6]
```diff
@@ -53,25 +62,39 @@ public Response getAllJobRoles(
...
+                    .getAllJobRoles(sanitisedPage, sanitisedPageSize,
+                            sanitisedFieldName, sanitisedOrderBy);
             Pagination pagination = new Pagination(
                     jobRoleService.getTotalpages(
                             sanitisedPageSize,
                             sanitisedPage),
                             sanitisedPage,
                    sanitisedPage + 1,
                 sanitisedPage - 1);
+            RoleOrdering roleOrdering = new RoleOrdering(
```
Comment (mlynch147) [inline src/main/java/org/example/controllers/JobRoleController.java:77]: Luke, just use the two values that you have here at lines 61 and 63.... then you can delete these two methods on the JobRoleService... they dont do anything else anyway, they just pass the string back out that you give it :)

File: src/main/java/org/example/models/RoleOrdering.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package org.example.models;
+
+public class RoleOrdering {
+    String fieldName;
```
Comment (mlynch147) [inline src/main/java/org/example/models/RoleOrdering.java]: make these two variables private

File: src/test/java/org/example/controller/JobRoleControllerTest.java
Comment (mlynch147) [inline src/test/java/org/example/controller/JobRoleControllerTest.java:1]: General comment - fix any tests that was mocking the methods that have been removed from JobRoleService

File: src/test/java/org/example/validators/OrderBySanitiserTest.java
Comment (mlynch147) [inline src/test/java/org/example/validators/OrderBySanitiserTest.java:1]: Can you add more 'valid' tests... eg one where orderBy is ASC, and then the others for the valid fieldNames

**Review by mlynch147 — CHANGES_REQUESTED** [review]
A few minor comments left throughout.... but it looks really good overall

**Review by mlynch147 — APPROVED** [review]

**Review by TaylorB-softdevkainos — APPROVED** [review]

**Pull request merged**